selection_tools/print_number_of_events.C

```diff
@@ -0,0 +1,196 @@
+    bkg_events_hists.push_back(bkg_events);
+    bkg_sigmas_hists.push_back(bkg_sigmas);
+    bkg_eff_hists.push_back(bkg_eff);
+    bkg_eff_sigmas_hists.push_back(bkg_eff_sigmas);
+  }
+
+  cout << "  ----------------  " << endl;
+  vector<double> sig_events;
+  vector<double> sig_sigmas;
+  vector<double> sig_eff;
+  vector<double> sig_eff_sigmas;
+  for(string hist_name : hist_names){
+    cout << hist_name << endl;
+    sig_events = {};
+    sig_sigmas = {};
+    sig_eff = {};
+    sig_eff_sigmas = {};
+    for(auto &[file_name_, params] : signal_files){
+      string file_name = file_name_;
+      file_name.erase(0,3);
+      cout << file_name << endl;
+      auto [color, title, signal, cross_sec, N_tot, BR] = params;
+      auto input_file = TFile::Open((base_path_sig+file_name).c_str());
+    
+      auto hist = (TH1D*)input_file->Get(hist_name.c_str());
+      
+      n_events = 0;
+      for(int i=0; i < hist->GetNbinsX()+1; i++)
+      {
+        n_events += hist->GetBinContent(i);
+      }
+      n_events_weighted = n_events*int_lumi*cross_sec*BR/N_tot;
+      n_events_sigma = sqrt(n_events)*int_lumi*cross_sec*BR/N_tot;
+      sig_events.push_back(n_events_weighted);
+      sig_sigmas.push_back(n_events_sigma);
+      sig_eff.push_back(100*n_events/N_tot);
+      sig_eff_sigmas.push_back(100*sqrt(n_events)/N_tot);
+    }
+    sig_events_hists.push_back(sig_events);
+    sig_sigmas_hists.push_back(sig_sigmas);
+    sig_eff_hists.push_back(sig_eff);
+    sig_eff_sigmas_hists.push_back(sig_eff_sigmas);
+  }
+
+  double bkg_n;
+  double bkg_sigma;
+  vector<double> s_start = sig_events_hists[0];
+  int j = 0;
+  for(auto &[file_name_, params] : signal_files){
+    auto [color, title, signal, cross_sec, N_tot, BR] = params;
+    std::cout << "---------------------- SIGNAL: " << title << " ---------------------- " << std::endl;
+    for(int i = 0; i<hist_names.size(); i++){
+      vector<double> s = sig_events_hists[i];
+      vector<double> s_sig = sig_sigmas_hists[i];
+      vector<double> s_eff = sig_eff_hists[i];
+      vector<double> s_eff_sig = sig_eff_sigmas_hists[i];
+      std::cout << cut_str[i] << "\t\t\t" << s[j] << "  +-  " << s_sig[j] << "\t\t\t" << s_eff[j] << "  +-  " << s_eff_sig[j] << std::endl;
+    }
+    j++;
+  }
+  // }
+
+  vector<double> b_start = bkg_events_hists[0];
+  j = 0;
+  for(auto &[file_name_, params] : background_files){
+    auto [color, title, signal, cross_sec, N_tot, BR] = params;
+    std::cout << "---------------------- BACKGROUND: " << title << " ---------------------- " << std::endl;
+    for(int i = 0; i<hist_names.size(); i++){
+      vector<double> b = bkg_events_hists[i];
+      vector<double> b_sig = bkg_sigmas_hists[i];
+      vector<double> b_eff = bkg_eff_hists[i];
+      vector<double> b_eff_sig = bkg_eff_sigmas_hists[i];
+      std::cout << cut_str[i] << "\t\t\t" << b[j] << "  +-  " << b_sig[j] << "\t\t\t" << b_eff[j] << "  +-  " << b_eff_sig[j] << std::endl;
+    }
+    j++;
+  }
+
+};
```

Comment:
Too many loops and operations in one block - should be split into small functions.